src/pages/patientView/SampleManager.tsx

```diff
@@ -269,6 +269,16 @@ class SampleManager {
         return this.samples.map((sample: ClinicalDataBySampleId) => sample.id);
     }
 
+    getSampleIdsIndex() {
+        let indexList: { [s: string]: number } = {};
+        this.samples
+            .map((sample: ClinicalDataBySampleId) => sample.id)
```

Comment:
negligible impact, but you don't need to do map.  you can just do samples.forEach(s=>indexList[s.sampleId]= ...)

Reply:
also, in general, when we transform something like this, it's best to use reduce because the code is clearest:
```const result = arr.reduce()```

Reply:
fixed